src/beachbot/robot/robotinterface.py

```diff
@@ -11,8 +20,122 @@ def __init__(self):
         self.platform=None
         self.arm=None
 
+        self.controller=None
+        self.controller_lock = threading.Lock()
+
+        self.detector=None
+        self.detector_lock = threading.Lock()
```

Comment:
Can you add some docstrings to these classes to explain how and where each of these locks are supposed to be used? You are adding a BUNCH of different locks. This can lead to all sorts of very difficult to find problems from just very easy to overlook changes. They should have clear intent of where they are used and what they are trying to sync (e.g. see my previous comment).